sympy/codegen/ast.py

```diff
@@ -1535,7 +1535,7 @@ def as_Declaration(self, **kwargs):
         >>> decl1.variable.value == NoneToken()  # OK
         True
         >>> decl2 = x.as_Declaration(value=42.0)
-        >>> decl2.variable.value == 42
+        >>> decl2.variable.value == 42.0
```

Comment:
I guess you don't have to fix this here, but I don't like doctests that are written like this. I would rather see

```py
>>> decl2.variable.value
42.0
```

We should only write doctests with `==` this when it's impossible to doctest the output for some reason (and even then, maybe just skipping the test can be better). 

Even when the output is `None` (which doctest hides), I'd rather see

```py
>>> print(x.is_integer)
None
```

than

```py
>>> x.is_integer == None
True
```

If these doctests were written this way in the first place you wouldn't need to fix them here.